cmd/metacache-entries.go

```diff
@@ -32,6 +32,14 @@ type metaCacheEntry struct {
 	// Entries without metadata will only be present in non-recursive scans.
 	metadata []byte
 
+	// versionIdx is generally set to '-1'
+	// it is only set for this entry if the version
+	// needs to be skipped by the caller. check relevant
+	// code in fileInfoVersions() on where this is used.
+	// once the version marker is consumed this value
+	// is set '-1'
+	versionIdx int
```

Comment:
maybe versionMarkerIndex is a better name

Reply:
it is a versionIdx instead of versionMarkerIndex marker is only top level, since its a lower-level call I kept it closer to what is meaningful from xl.meta